### DISCUSSION TITLE
GraphQL with WebFlux via GOM

### DISCUSSION THREAD

**Discussion by sp00m** [graphql-java/graphql-java#2482] [General]
Hey,

I just wanted to share back to the community a library we open-sourced: [GOM](https://github.com/qudini/gom). Built on top of [graphql-java](https://github.com/graphql-java/graphql-java), its purpose is to reduce the boilerplate needed to configure the schemas and the data fetchers/loaders.

It started in 2018: we wanted to give GraphQL a try for one of our Spring WebFlux microservices at [Qudini](https://www.qudini.com/). We originally started by using [graphql-java-tools](https://github.com/graphql-java-kickstart/graphql-java-tools), but the [`BatchedExecutionStrategy` got deprecated](https://github.com/graphql-java/graphql-java/issues/963) and [data loaders were not supported](https://github.com/graphql-java-kickstart/graphql-java-tools/issues/58). Because we couldn't accept _not_ being able to batch our IOs (plus we needed [further customisation of our SQL queries while batching](https://github.com/graphql-java/graphql-java/issues/1256) anyway), we decided to build our own abstraction, which eventually became GOM.

It covers _far_ from all that graphql-java does, and has room for improvements for sure, but it suits our needs quite well so far. It plays particularly well with Spring WebFlux too, see how we integrated it via [qudini-reactive-graphql](https://github.com/qudini/qudini-reactive/tree/master/qudini-reactive-graphql).

Don't hesitate to try it out, feedback is welcome, and thanks for your awesome work on graphql-java, keep it up 🚀

**Comment by bbakerman** [comment]
Great stuff.  I missed this somehow in my feed.  I was just reading it - well documented and looks like you have thought the API in depth.
